File: alphabit-coding-challenge-03/deadly_race/solutions/deadly_race.c
#include <stdio.h>
int main(){
  int N;
  scanf("%d",&N);
  int Ret = 2;
  while(1){
    if(N==3) break;  
    Ret*=2;
    N/=Ret%10;
  }
  printf("%d",Ret);
return 0;
}

File: alphabit-coding-challenge-03/deadly_race/solutions/deadly_race.java
import java.util.Scanner;

public class Race {

	public static void main(String[] args) {
		 int N;
			Scanner sc = new Scanner(System.in);		

		  N = sc.nextInt();;
		  int Ret = 2;
		  while(true){
		  if(N==3) break;  
		    Ret*=2;
		    N/=Ret%10;
		  }
		  System.out.print(Ret);
	}

}